assets/js/googlesitekit/widgets/datastore/widgets.js

```diff
@@ -159,7 +132,11 @@ export const reducer = ( state, { type, payload } ) => {
 				...state,
 				widgets: {
 					...state.widgets,
-					[ slug ]: { ...settings, slug },
+					[ slug ]: {
+						...settings,
+						component,
```

Comment:
See above, I think `component` should remain within `settings`, no need to split out.